### PR TITLE
Bug/dle 1013 fixes for transfer learning

### PR DESCRIPTION
**Pull request by shaydeci**
- Subclassing for unified Pascal VOC detection fixed.
- Attributes belonging to Detection|Dataset class added to the unified train set to support anchor generation.
- Downloading and organising Pascal data using UL script added.

### CODE REVIEW

File: src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py [hunk 1]
```diff
@@ -718,10 +723,20 @@ def __init__(self, dataset_params=None, cache_labels=False, cache_images=False):
         train_collate_fn = core_utils.get_param(self.dataset_params, 'train_collate_fn')
         val_collate_fn = core_utils.get_param(self.dataset_params, 'val_collate_fn')
 
+        class_inclusion_list = core_utils.get_param(self.dataset_params, "class_inclusion_list")
+        self.data_root = core_utils.get_param(self.dataset_params, 'data_root', "~/data/pascal_unified_coco_format/")
```
Comment (ofrimasad) [inline src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py:727]: mapping to home dir by default?

File: src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py [hunk 2]
```diff
@@ -732,10 +747,13 @@ def __init__(self, dataset_params=None, cache_labels=False, cache_images=False):
                                                          sample_loading_method=train_sample_method,
                                                          cache_labels=cache_labels,
                                                          cache_images=cache_images,
-                                                         augment=True)
+                                                         augment=True,
+                                                         class_inclusion_list=class_inclusion_list)
                 train_sets.append(sub_trainset)
```
Comment (ofrimasad) [inline src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py:752]: We are training on both train **and val**?

File: src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py [hunk 3]
```diff
@@ -746,10 +764,69 @@ def __init__(self, dataset_params=None, cache_labels=False, cache_images=False):
...
+
+        # ADDING USED ATTRIBUTES FROM PASCAL TO THE CONCATENATED DATASET
         self.trainset.classes = self.classes
+        self.trainset.img_size = self.dataset_params.train_image_size
+        self.trainset.img_files = img_files
+        self.trainset.cache_labels = cache_labels
+        self.trainset.exif_size = train_sets[1].exif_size
+        self.trainset.labels = labels
+
+    def download_pascal(self, delete=True):
```
Comment (ofrimasad) [inline src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py:783]: please rename to private _download_pascal

File: src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py [hunk 4]
```diff
@@ -746,10 +764,69 @@ def __init__(self, dataset_params=None, cache_labels=False, cache_images=False):
...
+        Source: https://github.com/ultralytics/yolov5/blob/master/data/VOC.yaml
+        """
+
+        def convert_label(path, lb_path, year, image_id):
+            def convert_box(size, box):
+                dw, dh = 1. / size[0], 1. / size[1]
+                x, y, w, h = (box[0] + box[1]) / 2.0 - 1, (box[2] + box[3]) / 2.0 - 1, box[1] - box[0], box[3] - box[2]
+                return x * dw, y * dh, w * dw, h * dh
+
+            in_file = open(path / f'VOC{year}/Annotations/{image_id}.xml')
```
Comment (ofrimasad) [inline src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py:798]: files are not properly closed. please use:
`with open(lb_path, 'w') as out_file`

File: src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py [hunk 5]
```diff
@@ -746,10 +764,69 @@ def __init__(self, dataset_params=None, cache_labels=False, cache_images=False):
...
+            tree = ET.parse(in_file)
+            root = tree.getroot()
+            size = root.find('size')
+            w = int(size.find('width').text)
+            h = int(size.find('height').text)
+            for obj in root.iter('object'):
+                cls = obj.find('name').text
+                if cls in PASCAL_VOC_2012_CLASSES and not int(obj.find('difficult').text) == 1:
+                    xmlbox = obj.find('bndbox')
+                    bb = convert_box((w, h), [float(xmlbox.find(x).text) for x in ('xmin', 'xmax', 'ymin', 'ymax')])
```
Comment (ofrimasad) [inline src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py:809]: please move the for loop into the function. and dont use x in the for loop (it is misleading)

File: src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py (hunk 4 above)
Comment (ofrimasad) [reply src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py:798]: and add 'path' into the string formating

**Review by ofrimasad — COMMENTED** [review]
the two functions:
download_and_unzip_from_url
download_pascal
needs to be rewritten

File: src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py (hunk 2 above)
Comment (shaydeci) [reply src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py:752]: Yes that's the common convention for VOC. Train on 2007 2012 trainval, and validate on 2007 test.

File: src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py (hunk 3 above)
Comment (shaydeci) [reply src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py]: renamed

File: src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py (hunk 4 above)
Comment (shaydeci) [reply src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py:798]: done

File: src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py (hunk 5 above)
Comment (shaydeci) [reply src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py:809]: done

File: src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py (hunk 1 above)
Comment (shaydeci) [reply src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py:727]: fixed

**Pull request merged**